store/src/java/com/zimbra/cs/event/logger/EventLogger.java

```diff
@@ -197,29 +200,33 @@ private EventNotifier(Map<String, EventLogHandler.Factory> knownFactories, Map<S
         @Override
         public void run() {
             try {
-                while (!Thread.currentThread().isInterrupted()) {
-                    consume(eventQueue);
+                while (!shutdownExecutor.get()) { //don't process any more events if shutdownExecutor is set to true.
+                    /* poll is used so we can check if shutdownExecutor is set every 3 seconds.
+                    if take was used the code would block on it in case of empty event queue. */
+                    Event event = eventQueue.poll(POLL_TIMEOUT, TimeUnit.SECONDS);
+                    if (event != null) {
+                         notifyEventLogHandlers(event);
+                    }
+                }
+                //drain the event queue till it is empty.
+                if (drainQueueBeforeShutdown.get()) {
+                    ZimbraLog.event.debug("Draining the queue");
+                    Event event = eventQueue.poll(POLL_TIMEOUT, TimeUnit.SECONDS);
```

Comment:
Could you interrupt the poll using Thread.interrupt() where shutdownExecutor is currently set? I do thing that if every restart during an install takes 3 seconds more that is a problem.

Reply:
While I am less concerned about installs - I think we should optimize the installer to avoid multiple restarts as if we are installing Windows 95 - I am concerned about delaying shutdown by 3 seconds. For scenarios where an admin just wants to shut down ZCS and does not care about events in the queue. Perhaps, you could make the timeout configurable via LC?